### PR TITLE
Improve credential errors

### PR DESCRIPTION
**Pull request by Marton6**
This PR, in combination with https://github.com/secrethub/secrethub-go/pull/223 ensures that all credential related errors are wrapped with a message that includes the source of the credential (except if the credential is passed via a cli flag).

### CODE REVIEW

File: internals/secrethub/credential_store.go [hunk 1]
```diff
@@ -69,10 +74,13 @@ func (store *credentialConfig) Import() (credentials.Key, error) {
 }
 
 func (store *credentialConfig) getCredentialReader() credentials.Reader {
-	if store.AccountCredential != "" {
-		return credentials.FromString(store.AccountCredential)
+	if store.AccountCredential == "" {
+		return store.configDir.Credential()
+	}
+	if store.credentialFlag.HasEnvarValue() {
+		return credentials.FromEnv("SECRETHUB_CREDENTIAL")
```
Comment (SimonBarendse) [inline internals/secrethub/credential_store.go:81]: It feels a bit awkward to have the flag already load the credential from the environment variable and validate it, but then load the credential from envvar again here and not use the already loaded value in `store.AccountCredential`.

We're doing all this to pass the source of the credential, did I get that right? If so, I'm wondering whether we can have the flag itself implement `CredentialSource` interface introduced in the accompanying PR https://github.com/secrethub/secrethub-go/pull/223 and return the flag itself as the credential reader?

The `Source` function of the credential flag could use `HasEnvarValue` to determine what to return.

Comment (Marton6) [reply internals/secrethub/credential_store.go]: Great idea! I implemented an (manually) tested it and it works well. The code is also much cleaner, thanks :+1:

**Comment by Marton6** [comment]
This PR now depends on https://github.com/secrethub/secrethub-go/pull/223 being merged first.

**Comment by Marton6** [comment]
PR was closed by mistake.

**Review by SimonBarendse — APPROVED** [review]

**Pull request merged**